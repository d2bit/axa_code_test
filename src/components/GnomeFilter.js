import React from 'react'
import styled from 'styled-components'

const Frame = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  height: 50px;
  background-color: white;
  border-bottom: 1px solid #d2d2d2;
  width: 100%;
  padding-top: .5rem;
`
const Label = styled.label`
  padding: 0 1em 0;
`
const Input = styled.input`
  font-size: 1.5rem;
  padding: .2rem .5rem;
  border: 0;
  border-bottom: .5px solid #efefef;
`

const GnomeFilter = ({ filter, updater }) => {
  const handleChange = ev => {
    ev.preventDefault()
    const { value } = ev.target

    updater({ name: value })
  }

  const { name } = filter

  return (
    <Frame>
      <Label htmlFor='GnomeFilter'>Filter</Label>
      <Input data-testid='GnomeFilter' name='GnomeFilter' value={name} onChange={handleChange} />
    </Frame>
  )
}

export default GnomeFilter
