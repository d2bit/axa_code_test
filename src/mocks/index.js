import data from './data.json'

const population = Object.values(data)[0]
const populationCount = population.length

export function getPopulation(size=5) {
  if (size >= populationCount) {
    return population
  }
  return population.slice(0, size)
}

export function getGnome(index) {
  if (index && index >= 0 && index < populationCount) {
    return population[index]
  }
  const randomIndex = Math.floor(Math.random() * populationCount)
  return population[randomIndex]
}

export function getResponseData() {
  return data
}

const Helper = {
  getPopulation,
  getGnome,
  getResponseData,
}

export default Helper
