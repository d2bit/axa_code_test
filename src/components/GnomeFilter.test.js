import React from 'react'
import { render, Simulate, wait } from 'react-testing-library'
import 'dom-testing-library/extend-expect'
import { getGnome } from '../mocks'
import GnomeFilter from './GnomeFilter'

describe('GnomeFilter', () => {
  it('shows the gnome information', () => {
    const updaterFn = jest.fn()
    const filter = { name: '' }

    const { getByTestId, container } = render(<GnomeFilter filter={filter} updater={updaterFn} />)
    expect(container).toMatchSnapshot()

    const filterInput = getByTestId('GnomeFilter')
    Simulate.change(filterInput, { target: { value: 'P' } })
    expect(updaterFn).toHaveBeenCalledTimes(1)
    expect(updaterFn).toHaveBeenCalledWith({ name: 'P' })
  })
})
