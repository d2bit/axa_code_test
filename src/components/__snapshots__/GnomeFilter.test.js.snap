// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GnomeFilter shows the gnome information 1`] = `
<div>
  <div
    class="sc-bdVaJa dtiqAs"
  >
    <label
      class="sc-bwzfXH jxEbzC"
      for="GnomeFilter"
    >
      Filter
    </label>
    <input
      class="sc-htpNat fKBxqF"
      data-testid="GnomeFilter"
      name="GnomeFilter"
      value=""
    />
  </div>
</div>
`;
